test/gpu/native/gpuWritelnAndAssertOnGpu.chpl

```diff
@@ -0,0 +1,33 @@
+use GPU;
+
+proc subFunc() {
+  gpuWriteln("hello from function called by GPU!".c_str());
+  gpuWriteln("hello again from ".c_str(), " function called by GPU!".c_str());
+}
+
+writeln("Writeln from CPU");
+on here.gpus[0] {
+  var A : [0..10] int;
+  foreach i in 0..10 {
+    A[i] = i;
+    assertOnGpu();
+    if(i == 0) {
+      gpuWriteln("hello from the GPU!".c_str());
+      gpuWriteln("hello from the GPU ".c_str(), "again!".c_str());
+      subFunc();
+    }
+  }
+  // At this point the kernel should end and everything coming from gpuWriteln
+  // should have been flushed to the terminal..
```

Comment:
Just a question. If the kernel segfaults before ending, are the `gpuWriteln`s still flushed to the console?

Reply:
I haven't tested it but I think not. In the end the runtime is just calling out to the same `printf` that you would use in CUDA (so it gets buffered than flushed after the kernel finishes in the same way).

Of course now that I say that I'm confused why we didn't run into the same "you overwrote the output buffer and thus a lot of it gets cut off when it finally flushes" problem in your Chapel code that we were seeing in the CUDA one.

We were just calling into the runtime library to use using printf there too right?

Reply:
I think we didn't run into that because we were printing to `stderr` which gets flushed more often.

Reply:
Yeah but I assume that still gets buffered and flushed at the end of the kernel? Maybe the buffer size is different by default or maybe CUDA really is doing some kind of magic to get it back to the CPU in real time?